app/src/main/java/com/example/dine_and_donate/EventActivity.java

```diff
@@ -158,29 +166,42 @@ public void onComplete(@NonNull Task<Uri> task) {
                                 downloadUri[0] = task.getResult();
                                 String s = downloadUri[0].toString();
                                 writeEvent(intent, s, location);
-                            } else {
-                                // Handle failures
                             }
                         }
                     });
-
                 }
             }
         });
     }
 
     private void writeEvent(Intent intent, String url, String location) {
         String orgId = mFirebaseCurrentUser.getUid();
-        String yelpId = intent.getStringExtra("yelpID");
+        final String yelpId = intent.getStringExtra("yelpID");
         long eventDate = mCalendarView.getDate();
         long startTime = convert(eventDate, mStartHour.getSelectedItemPosition()+1, mStartMin.getSelectedItemPosition(), mStartHalf.getSelectedItem().equals("PM"));
         long endTime = convert(eventDate, mEndHour.getSelectedItemPosition()+1, mEndMin.getSelectedItemPosition(), mEndHalf.getSelectedItem().equals("PM"));
         String info = mEtEventInfo.getText().toString();
-        Event newEvent = new Event(orgId, yelpId, location, startTime, endTime, info, url);
-        mRef.child("events").child(yelpId).child(UUID.randomUUID().toString()).setValue(newEvent);
+        newEvent = new Event(orgId, yelpId, location, startTime, endTime, info, url);
+        mRef.child("events").child(yelpId).child(UUID.randomUUID().toString()).setValue(newEvent, new DatabaseReference.CompletionListener() {
+            @Override
+            public void onComplete(@Nullable DatabaseError databaseError, @NonNull DatabaseReference databaseReference) {
+                //there is an error, can not add event to database
+                if (databaseError != null) {
+                    Toast.makeText(EventActivity.this, "Error Saving Data To Database", Toast.LENGTH_LONG).show();
+                } else {
+                    mCreatedEvents.put(databaseReference.getKey(), yelpId);
+                    mRef.child("users").child(mFirebaseCurrentUser.getUid()).child("Events").setValue(mCreatedEvents);
+                    Toast.makeText(EventActivity.this, "Saved Data To Database", Toast.LENGTH_LONG).show();
```

Comment:
I don't think we want to show a message "Saved Data to Database" to the user every time it's saved. I would delete this